Firestore/Protos/protos/google/firestore/v1beta1/firestore.options

```diff
@@ -0,0 +1,6 @@
+google.firestore.v1beta1.GetDocumentRequest.consistency_selector no_unions:true
```

Comment:
> Did you write these files?

Yes.

> Is there no way to make this just a global option?

Not that I'm aware of. But this isn't terribly well documented, so perhaps there is a way to do this? OTOH, you might not want to apply this globally: oneofs without variable length fields are supported without this option being enabled just fine (and would be more memory efficient too. With this option enabled, nanopb generates each field as an optional field rather than as a union, thus taking up more memory.)

> As it stands this seems fragile: any addition to the protocol has the potential to break our generated code. If we fail to add a no_unions:true for a oneof, what happens?

Assuming the oneof has a variable length field, the protos will fail to build with an error message like this:
```
Traceback (most recent call last):
<python stack trace>
Exception: Callback fields inside of oneof are not supported (field string_value)
--nanopb_out: protoc-gen-nanopb: Plugin failed with status code 1.
```

In this case "string_value" is the name of the first variable length field within the oneof.

Reply:
I see. So if we miss a value we needed to do this on we'd know pretty fast. Seems okay then.